test-framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -562,14 +555,22 @@ private static String groupName(ThreadGroup threadGroup) {
      * Returns size random values
      */
     public static <T> List<T> randomSubsetOf(int size, T... values) {
-        if (size > values.length) {
-            throw new IllegalArgumentException("Can\'t pick " + size + " random objects from a list of " + values.length + " objects");
+        return randomSubsetOf(size, Arrays.asList(values));
+    }
+
+    /**
+     * Returns size random values
+     */
+    public static <T> List<T> randomSubsetOf(int size, Collection<T> values) {
+        if (size > values.size()) {
+            throw new IllegalArgumentException("Can\'t pick " + size + " random objects from a list of " + values.size() + " objects");
         }
-        List<T> list = arrayAsArrayList(values);
```

Comment:
There is an unused import `import static org.elasticsearch.common.util.CollectionUtils.arrayAsArrayList;` that can be dropped now.